### PR TITLE
Add media filters

### PR DESCRIPTION
**Pull request by kean**
This is something that was originally suggested in the Modernization spec, but I didn't have time to implement, so here it is now.

- Add media filters
- Add “Aspect Fill Grid” support to iPhone

Not in the scope: changing the empty state; seems acceptable.

To test:

- Verify that the navigation title now has a menu that allows you to switch between media types and also change the display mode: aspect/square
- Verify that the aspect ratio display mode is now available on iPhone
- Verify that the selection is **not** persisted
- Verify that when you make the selection, the navigation title also updates 

Screenshot 2023-11-27 at 8 55 03 AM Screenshot 2023-11-27 at 8 55 08 AM

Screenshot 2023-11-27 at 8 57 40 AM

## Regression Notes
1. Potential unintended areas of impact: Site Media
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/Extensions/UIButton+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import UIKit
+
+extension UIButton {
+    /// Creates a bar button item that looks like the native title menu
+    /// (see `navigationItem.titleMenuProvider`, iOS 16+).
+    static func makeMenuItem(title: String, menu: UIMenu) -> UIButton {
```
Comment (momo-ozawa) [inline WordPress/Classes/Extensions/UIButton+Extensions.swift]: Wdyt of renaming this to makeFilterButton or makeDropDownButton? MenuItem makes me think of an individual item in a menu

File: WordPress/Classes/ViewRelated/Media/SiteMedia/SiteMediaViewController.swift [hunk 2]
```diff
@@ -287,7 +299,7 @@ extension SiteMediaViewController {
 }
 
 private enum Strings {
-    static let title = NSLocalizedString("mediaLibrary.title", value: "Media", comment: "Media screen navigation title")
+        static let title = NSLocalizedString("mediaLibrary.title", value: "Media", comment: "Media screen navigation title")
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Media/SiteMedia/SiteMediaViewController.swift]:
```suggestion
    static let title = NSLocalizedString("mediaLibrary.title", value: "Media", comment: "Media screen navigation title")
```

**Review by momo-ozawa — APPROVED** [review]
LGTM w 2 small nits ✨

File: WordPress/Classes/Extensions/UIButton+Extensions.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/Extensions/UIButton+Extensions.swift]: Oh, I forgot to rename it. It was originally an extension for `UIBarButtonItem` and was returning an instance of it. What do you think about `UIButton.makeMenu`?

Comment (momo-ozawa) [reply WordPress/Classes/Extensions/UIButton+Extensions.swift]: `UIButton.makeMenu` sounds good!

**Pull request merged**